jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/SessionHandler.java

```diff
@@ -445,29 +497,98 @@ public void setUsingUriParameters(boolean value)
     @Override
     public void doScope(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
     {
-        if (baseRequest.getDispatcherType() == DispatcherType.REQUEST)
-        {
-            ContextHandler.CoreContextRequest coreRequest = baseRequest.getHttpChannel().getCoreRequest();
+        ContextHandler.CoreContextRequest coreRequest = baseRequest.getHttpChannel().getCoreRequest();
+        SessionManager oldSessionManager = null;
+        ManagedSession oldSession = null;
+        AbstractSessionManager.RequestedSession oldRequestedSession = null;
 
-            // TODO should we use the Stream wrapper? Could use the HttpChannel#onCompleted mechanism instead.
-            _sessionManager.addSessionStreamWrapper(coreRequest);
+        ManagedSession currentSession = null;
+        AbstractSessionManager.RequestedSession currentRequestedSession = null;
 
-            // find and set the session if one exists
-            AbstractSessionManager.RequestedSession requestedSession = _sessionManager.resolveRequestedSessionId(coreRequest);
-
-            coreRequest.setSessionManager(_sessionManager);
-            coreRequest.setRequestedSession(requestedSession);
+        try
+        {
+            switch (baseRequest.getDispatcherType())
+            {
+                case REQUEST:
+                {
+                    _sessionManager.addSessionStreamWrapper(coreRequest);
```

Comment:
What if the request landed on an EE10 context first and is then forwarded to EE9.  Won't that result in the core request having a session (managed by the normal session stream wrapper) and the EE9 Request will have another session that will need to be managed by this stream wrapper?

Reply:
Yes, if a request created/loaded a session in an ee10 context, then that session lifecycle is managed by the core SessionStreamWrapper. If that request is then forwarded to an ee9 context, then the dispatch type is not REQUEST, so we skip adding a SessionStreamWrapper. So I think that's a problem as a session created/loaded whilst in the ee9 dispatch won't be completed as the request exists. Thinking about a solution ...